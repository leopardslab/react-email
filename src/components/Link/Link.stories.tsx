import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Link } from './Link';

export default {
  component: Link,
} as ComponentMeta<typeof Link>;

//“template” of how args map to rendering
const Template: ComponentStory<typeof Link> = (args) => (
  <table>
    <tbody>
      <tr>
        <Link {...args} />
      </tr>
    </tbody>
  </table>
);

export const Default = Template.bind({});

Default.args = {
  children: 'Click here',
  href: 'https://github.com/leopardslab/react-email',
};
