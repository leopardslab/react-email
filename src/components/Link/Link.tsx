import { ReactNode, HTMLAttributeAnchorTarget } from 'react';
import type { BaseStyleProp } from '../types';
import { makeStyles } from '../../utils/makeStyles';
import { useTheme } from '../ThemeProvider';

export type LinkStyles = 'root';

/**
 * Interface for PropTypes for the `Link` component.
 */
export interface LinkProps extends BaseStyleProp<LinkStyles> {
  /**
   * The content of the component.
   */
  children?: ReactNode;
  /**
   * The URL to link to.
   */
  href: string;
  /**
   * The target of the link.
   * @default '_blank'
   */
  target?: HTMLAttributeAnchorTarget;
}

const useStyles = makeStyles({
  root: {},
});

export const Link = ({
  children,
  href,
  target = '_blank',
  classes,
  className,
}: LinkProps): JSX.Element => {
  const themeClasses = useTheme('link');
  const styles = useStyles({ classes }, themeClasses);

  return (
    <a href={href} target={target} style={styles.root} className={className}>
      {children}
    </a>
  );
};
