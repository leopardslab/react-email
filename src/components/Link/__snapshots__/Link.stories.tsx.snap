// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Link CustomStyles smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px;">
  <table cellpadding="0"
         cellspacing="0"
         role="presentation"
         style="border: 0px; vertical-align: top; width: 100%;"
  >
    <tbody>
      <tr>
        <td align="center"
            style="vertical-align: top;"
        >
          <h2>
            Please
            <a href="https://github.com/leopardslab/react-email"
               target="_blank"
               style="color: red; text-decoration: underline;"
            >
              Click here
            </a>
            to access
          </h2>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`components/Link Default smoke-test 1`] = `
<div style="margin: 0px auto; max-width: 600px;">
  <table cellpadding="0"
         cellspacing="0"
         role="presentation"
         style="border: 0px; vertical-align: top; width: 100%;"
  >
    <tbody>
      <tr>
        <td align="center"
            style="vertical-align: top;"
        >
          <h2>
            Please
            <a href="https://github.com/leopardslab/react-email"
               target="_blank"
               style="color: rgb(0, 0, 238); text-decoration: none;"
            >
              Click here
            </a>
            to access
          </h2>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
